contrib/cirrus/setup_environment.sh

```diff
@@ -54,6 +54,10 @@ if [[ "$CI_DESIRED_STORAGE" = "composefs" ]]; then
     if [[ "$PRIV_NAME" == "root" ]]; then
         CI_DESIRED_COMPOSEFS="+composefs"
 
+        # artifacts restored from a tarball won't be stored correctly
+        # in composefs, so we need to disable this feature.
```

Comment:
Is this a "will never be stored correctly", or "there's a bug that we need to fix"? Future maintainers will need to know this. (And, if the latter, will want an issue number).

Also, I suggest renaming the variable to `CI_NO_TEST_ARTIFACTS` or something with a `CI_` prefix. A `CI_` prefix is treated specially, such that it gets propagated to all test environments (like rootless). I know composefs is root-only, but maybe some day that will change, or maybe this NO_ARTIFACT thing will get reused for a different purpose some day, and whoever deals with that may spend a lot of time wondering why rootless isn't working. Naming it properly today will be a silent gift to that person.

Reply:
I used a named similar to the existing `NO_TEST_CACHE`.  Renamed to `CI_NO_TEST_ARTIFACTS`.

Reply:
I think the question is still open here, is this temporary or permanent?